sqlglot/generator.py

```diff
@@ -3820,7 +3820,7 @@ def copy_sql(self, expression: exp.Copy) -> str:
         this = f" INTO {this}" if self.COPY_HAS_INTO_KEYWORD else f" {this}"
 
         credentials = self.sql(expression, "credentials")
-        credentials = f" {credentials}" if credentials else ""
+        credentials = f"{credentials}" if credentials else ""
```

Comment:
why was this space removed? if it's not needed, you don't need the f sting

Reply:
Pushed the spaces down to `credentials_sql()` as there was a spacing bug with a certain combination of creds and was testing the result there, thanks for the catch